tape/src/main/java/com/squareup/tape/QueueFile.java

```diff
@@ -182,8 +201,8 @@ private void readHeader() throws IOException {
     elementCount = readInt(buffer, 4);
     int firstOffset = readInt(buffer, 8);
     int lastOffset = readInt(buffer, 12);
-    first = readElement(firstOffset);
-    last = readElement(lastOffset);
+    first = readElement(firstOffset, true);
+    last = readElement(lastOffset, true);
```

Comment:
since this is all single threaded, you could also make this a field boolean set to true then false instead of plumbing it through.

Reply:
seems more complex for future maintenance.